src/browser/api_protocol/api_handlers/mesh_middleware.ts

```diff
@@ -128,6 +134,84 @@ function sendMessageMiddleware(msg: MessagePackage, next: () => void) {
     }
 }
 
+const handleExternalWindow = async (identity: Identity, toGroup: Identity) => {
+    const { uuid, name } = toGroup;
+    if (coreState.getWindowByUuidName(identity.uuid, name)) {
+        // External window already created and registered
+        return 'registered';
+    }
+    if (!uuid || isLocalUuid(uuid)) {
+        return;
+    }
+    try {
+        const getHwndMessage = {
+            action: 'get-window-native-id',
+            payload: {
+                uuid,
+                name
+            }
+        };
+        const id = await connectionManager.resolveIdentity({uuid});
+        const hwnd = await id.runtime.fin.System.executeOnRemote(identity, getHwndMessage);
+
+        const childWindowOptions = {
+            uuid: identity.uuid,
+            name,
+            hwnd: hwnd.data
+        };
+        const parent = coreState.getWindowByUuidName(identity.uuid, identity.name);
+        const parentId = parent && parent.browserWindow && parent.browserWindow.id;
+        const childBw = new BrowserWindow(childWindowOptions);
+        const childId = childBw && childBw.id;
+
+        if (!coreState.addChildToWin(parentId, childId)) {
+            console.warn('failed to add child window');
+        }
+        Window.create(childId, childWindowOptions);
+
+        return hwnd;
+    } catch (e) {
+        log.writeToLog('info', e);
+        return e;
+    }
+};
+
+async function meshJoinWindowGroupMiddleware(msg: MessagePackage, next: (locals?: object) => void) {
```

Comment:
Should type out return, like `: Promise<void>`...

Reply:
:+1: